dev/storage.h

```diff
@@ -2923,6 +2935,63 @@ namespace sqlite_orm {
         public:
             pragma_type pragma;
             limit_accesor<self> limit;
+
+        private:
+            /*
+             * Delete routine
+             * O is an object's type. Must be not specified explicitly cause it can be deduced by
+             *      compiler from first parameter.
+             * @param o object to be updated.
+             */
+            template<class O>
+            void remove_by_object_impl(const O &o, std::false_type) {
+                this->assert_mapped_type<O>();
+
+                auto &impl = this->get_impl<O>();
+
+                std::vector<std::string> primaryKeyColumnNames = impl.table.primary_key_column_names();
+
+                impl.table.for_each_column([this,&o,&primaryKeyColumnNames] (auto &c) {
+					if(std::find(primaryKeyColumnNames.cbegin(), primaryKeyColumnNames.cend(), c.name) !=
+                            primaryKeyColumnNames.cend()) {
+                        using field_type = typename std::decay<decltype(c)>::type::field_type;
+                        const field_type *value = nullptr;
+                        if(c.member_pointer){
+                            value = &(o.*c.member_pointer);
+                        }else{
+                            value = &((o).*(c.getter))();
+                        }
+                        this->remove<O>(*value);
+                    }
+                });
+            }
+
+            /*
+             * Delete routine
+             * O is an object's type. Must be not specified explicitly cause it can be deduced by
+             *      compiler from first parameter.
+             * @param o object to be updated.
+             */
+            template<class O>
+            void remove_by_object_impl(const O &o, std::true_type) {
+                this->assert_mapped_type<O>();
+
+                auto &impl = this->get_impl<O>();
+
+				typedef typename decltype(impl.table)::impl_type::composite_key_type::columns_type composite_key_columns_type;
+				typedef typename decltype(impl.table)::impl_type::composite_key_type::columns_value_type composite_key_columns_value_type;
+				composite_key_columns_type composite_key_columns = impl.table.get_composite_key().columns;
+				composite_key_columns_value_type composite_key_value;
+
+				tuple_helper::apply([&o,&composite_key_value](auto ...pks) {
```

Comment:
@fnc12 what is meaning of copying arguments?

Reply:
Use `(auto ...&pks)` instead of `(auto ...pks)`

Reply:
@fnc12 Ok I know it, but as I said before if I change it to `(auto...&pks)` it cause compile error.

Reply:
What error?

Reply:
`error: invalid use of non-static member function of type ‘int (testRemove()::Object::)() const`

Reply:
@fnc12 I don't think copy of some member pointer functions is very performance critical, if you agree with me, we can resolve this conversation, or if you have any idea to solve compile error, I very welcome you :)

Reply:
Yes, copying of member pointer is a very cheap operation. I think that there is a way to avoid using tuple::apple and not to add it to the lib cause it will be included in C++17 as far as I know

Reply:
So I resolved this conversation